src/Common/src/Common/Contexts/AbstractApplicationContext.cs

```diff
@@ -20,10 +20,17 @@ public abstract class AbstractApplicationContext : IApplicationContext
     {
         private readonly ConcurrentDictionary<string, object> _instances = new ConcurrentDictionary<string, object>();
 
-        protected AbstractApplicationContext(IServiceProvider serviceProvider, IConfiguration configuration)
+        protected AbstractApplicationContext(IServiceProvider serviceProvider, IConfiguration configuration, IEnumerable<NameToTypeMapping> nameToTypeMappings)
         {
             ServiceProvider = serviceProvider;
             Configuration = configuration;
+            if (nameToTypeMappings != null)
```

Comment:
If it is normal/ok for `nameToTypeMappings` to be null, should the param be nullable too? Alternatively, should it always be provided and an exception thrown if null?

Reply:
It is normal to be nullable- derived class passes null by default-so no exception. If you are asking if it should default to null, its fine either way. The param is nullable.